### ISSUE TITLE
SP equivalent

### ISSUE THREAD

**Issue by williamfromIT** [OfficeDev/office-scripts-docs#424]
[Enter feedback here]
Can we have a SharePoint equivalent please?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 688cb5b0-672f-8480-8943-6e4bc403d6b9
* Version Independent ID: 61518540-2796-b66f-4bc7-aef0eca420ad
* Content: [Combine workbooks into a single workbook - Office Scripts](https://docs.microsoft.com/en-us/office/dev/scripts/resources/samples/combine-worksheets-into-single-workbook)
* Content Source: [docs/resources/samples/combine-worksheets-into-single-workbook.md](https://github.com/OfficeDev/office-scripts-docs/blob/master/docs/resources/samples/combine-worksheets-into-single-workbook.md)
* Product: **excel**
* Technology: **scripts**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by AlexJerabek** [comment]
Hi @williamfromIT,
I'll investigate adding that case to the sample. In the meantime, can you please elaborate more on your scenario? What is the workflow you're trying to automate?
Thanks!

**Comment by williamfromIT** [comment]
Hi @AlexJerabek 

On my SP I have a folder called Customer containing 10 folders with each customer name. The 10 folders contain 2 - 5 .xlsx files each.

My goal is to get all of those different .xlsx files into a single customer file.

**Comment by peronald** [comment]
@williamfromIT you can achieve your scenario today by choosing the SharePoint file in the Excel Online connector. Instead of OneDrive, you would choose the site of the workbook in the Location property of the connector and then select the relevant Document Library and File path of the workbook that is located in the SharePoint site you chose.

**Comment by williamfromIT** [comment]
@AlexJerabek @peronald 

Thank you!

Hi guys! I got it half way made. It takes the first report and places it in my newly created document but then it hangs and the flow does not finish. It does not process the rest of the files. 
I attached some pictures. [Link](https://imgur.com/a/5Tevtg5)

**Comment by williamfromIT** [comment]
@AlexJerabek @peronald 

What is the sheet size of what can be returned on a sheet? I think my 10K is to much?

**Comment by AlexJerabek** [comment]
Hi @williamfromIT,

Here are the [data limits](https://docs.microsoft.com/en-us/office/dev/scripts/testing/platform-limits#data-limits) for Office Scripts with Excel on the web:
- Requests and responses are limited to 5MB.
- A range is limited to five million cells. 

Here's the advice about dealing with large data sets from that article:

> If you're encountering errors when dealing with large datasets, try using multiple smaller ranges instead of larger ranges. For an example, see the [Write a large dataset sample](https://docs.microsoft.com/en-us/office/dev/scripts/resources/samples/write-large-dataset). You can also use APIs like [Range.getSpecialCells](https://docs.microsoft.com/en-us/javascript/api/office-scripts/excelscript/excelscript.range?view=office-scripts#getSpecialCells_cellType__cellValueType_) to target specific cells instead of large ranges.

**Comment by williamfromIT** [comment]
I am having an issue. The script works to return sheets with 250K cells, no problem. on 350K+ it sits and never completes. Well under the 5 million data limits. Any ideas?

**Comment by AlexJerabek** [comment]
@williamfromIT, it's possible you're reaching the [5MB limit for requests and responses](https://docs.microsoft.com/en-us/office/dev/scripts/testing/platform-limits#excel). If that's the case, an individual script would fail outside of Power Automate.

The [Write a large dataset sample](https://docs.microsoft.com/en-us/office/dev/scripts/resources/samples/write-large-dataset) provides an example for splitting a large write command into multiple steps. If that doesn't help unblock you, please either create a new issue or ask a question on [Stack Overflow with the office-scripts tag](https://stackoverflow.com/questions/tagged/office-scripts). Stack Overflow is a great place to get community support for questions like this.
